### PR TITLE
Add a timeout for GCE detection.

### PR DESCRIPTION
**Pull request by craigcitro**
This is a temporary workaround for #93, though may end up being the final fix.

I did some test cleanup while I was here, switching to `mock`.

PTAL @dhermes

### CODE REVIEW

File: oauth2client/client.py [hunk 1]
```diff
@@ -930,12 +931,22 @@ def _detect_gce_environment(urlopen=None):
           Compute Engine.
   """
   urlopen = urlopen or urllib.request.urlopen
-
+  # Note: the explicit `timeout` below is a potentially-short-term
```
Comment (dhermes) [inline oauth2client/client.py]: Pretty silly to call it short-term and not give a date. I'd suggest either not mentioning that it's short term or giving it a timestamp (I know a `git blame` or other could locate the time, but people won't do that).

Comment (craigcitro) [inline oauth2client/client.py]: s'fair. dropped "potentially-short-term".

File: oauth2client/client.py [hunk 2]
```diff
@@ -930,12 +931,22 @@ def _detect_gce_environment(urlopen=None):
...
+  # workaround. The underlying issue is that resolving an unknown host
+  # on some networks will take 20-30 seconds; making this timeout
+  # short fixes the issue, but could lead to false negatives in the
+  # event that we are on GCE, but the metadata resolution was
+  # particularly slow. The latter case is "unlikely".
   try:
-    response = urlopen('http://metadata.google.internal')
+    response = urlopen('http://metadata.google.internal/', timeout=1)
```
Comment (dhermes) [inline oauth2client/client.py:941]: It just occurred to me that only headers are necessary. Worth filing a bug to make this a `HEAD` request?

File: tox.ini [hunk 3]
```diff
@@ -4,6 +4,7 @@ envlist = py26,py27,py33,py34,pypy,cover
 [testenv]
 basedeps = keyring
            mox3
+           mock
```
Comment (dhermes) [inline tox.ini:7]: Mox and mock? At the very least you should file an admittedly aspirational bug to choose one ~~ring~~ mocking library to rule them all.

File: tests/test_oauth2client.py [hunk 4]
```diff
@@ -208,37 +206,19 @@ def test_get_environment_gae_local(self):
...
-    urllib2_urlopen.__call__(('http://metadata.google.internal'
-                             )).AndReturn(mockResponse)
-
-    m.ReplayAll()
-
-    self.assertEqual('GCE_PRODUCTION', _get_environment(urllib2_urlopen))
-
-    m.UnsetStubs()
-    m.VerifyAll()
+    with mock.patch.object(urllib.request, 'urlopen') as urlopen:
```
Comment (dhermes) [inline tests/test_oauth2client.py:209]: Sorry for being a weenie but why are these necessary? What's wrong with `mox`?

File: oauth2client/client.py (hunk 2 above)
Comment (craigcitro) [inline oauth2client/client.py:941]: we could -- but the response would only be large in the case of something like a captive portal, where i'm downloading a whole page, right? in that case, might be more work than it's worth.

Comment (dhermes) [inline oauth2client/client.py:941]: Yes after writing I thought "probably more work than it's worth" :)

File: tox.ini (hunk 3 above)
Comment (craigcitro) [inline tox.ini:7]: #102

File: tests/test_oauth2client.py (hunk 4 above)
Comment (craigcitro) [inline tests/test_oauth2client.py:209]: my brain can also only keep one mocking library loaded, and `mock` seems to be the future (it's in the stdlib as of 3.3). might as well start moving that way.

Comment (dhermes) [inline tests/test_oauth2client.py:209]: It's your mess, but that seems like a terrible idea without a clear plan.

Comment (craigcitro) [inline tests/test_oauth2client.py:209]: going to push a change to drop mox shortly. otherwise anything else on this one?
